storage.go

```diff
@@ -34,28 +34,43 @@ func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
 func (s *Storage) createAppend(ctx context.Context, path string, opt pairStorageCreateAppend) (o *Object, err error) {
 	rp := s.getAbsPath(path)
 	dir := filepath.Dir(rp)
-	err = s.hdfs.MkdirAll(dir, 0666)
+
+	//	If dirname is already a directory,
+	// 	MkdirAll does nothing and returns nil.
+	err = s.hdfs.MkdirAll(dir, 0755)
+	//	If dirname is not exist ,it will create a Mkdir rpc communication
+	//	So we just need to catch other errors
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	_, err = s.hdfs.Stat(rp)
 
+	//	The error returned by Stat can only be nil or not os.ErrNotExist
 	if err == nil {
-		err = s.hdfs.Remove(rp)
-		if err != nil && errors.Is(err, os.ErrNotExist) {
-			// Omit `file not exist` error here
-			// ref: [GSP-46](https://github.com/beyondstorage/specs/blob/master/rfcs/46-idempotent-delete.md)
-			err = nil
+		// If the path does not exist,
```

Comment:
If the error returned by `Stat` is nil, the path must exist. I think we don't need to explain the detailed behavior of our hdfs SDK. Instead, we need to comment on why we need to do remove it here.